lua/lazyvim/plugins/extras/linting/golangci-lint.lua

```diff
@@ -0,0 +1,31 @@
+return {
+  {
+    "mason.nvim",
+    opts = function(_, opts)
+      vim.list_extend(opts.ensure_installed, {
+        "golangci-lint",
+      })
+    end,
+  },
+
+  {
+    "mfussenegger/nvim-lint",
+    optional = true,
+    opts = {
+      linters_by_ft = {
+        go = { "golangcilint" },
```

Comment:
```suggestion
        go = { "golangcilint" },
        gomod = { "golangcilint" },
        gowork = { "golangcilint" },
```
Since it works for `go.mod` and `go.work` too.